### ISSUE TITLE
correctly updating the server URL in account settings messes up local database

### ISSUE THREAD

**Issue by unDocUMeantIt** [haiwen/seafile-client#1321]
# background

i had to change the URL of a seafile server, i.e., the same server continued to run but under a different subdomain. let's call them `https://old.example.com:1234/seafile` and `https://new.example.com:1234/seafile` (indicating that both port and local subdirectory weren't the default, although it is unknown if this is relevant). after successfully changing the server configuration accordingly, so it was accessible via webbrowser, i began to reconfigure my seafile clients: the account settings offer to change a server's URL.

# expected results

i expected the client to successfully update its configuration to communicate with the server by its new address.

# what actually happened

most communication with the server was lost; all synced libraries were shown as grey clouds, i was unable to manually sync, force a resync or even stop syncing any of those libraries. i could, however, open all of them in both the webbrowser and the client's internal file browser. neither logging out and back in again, nor fully removing the account and adding it back again made any difference. i was stuck with an unusable client. this happenend on two different computers.

## examining the local database files

when i examined the local database files in the `.seafile-data` directory, i noticed that some database cells were obviously updated with incomplete data. in particular, instead of `https://new.example.com:1234` i found `new.example.com` in some tables, as well as a configured `relay-address` pointing to the outdated URL. it looks as if the client failed at updating the databases correctly, omitting transport protocol and port of the new URL.

# manual workaround

this is how i was able to fix the issue by manually(!) fixing the database files:

 1. if present, remove an incomplete URL from table `ServerProperty` of database `repo.db`

 2. set all instances of `relay-address` pointing to the old URL to `NULL` in table `RepoProperty`of database `repo.db`

 3. fix all instances of `server-url`, from `new.example.com` into `https://new.example.com:1234` in table `RepoProperty`of database `repo.db`

 4. launch the client and -- if neccessary -- demand a full resync of all libraries from that server

**Comment by killing** [comment]
Which version of Seafile client do you use?

**Comment by unDocUMeantIt** [comment]
you're right, i forgot: it's 7.0.6, the version that is shipped with ubuntu 20.04.

this report is a summary of an earlier [forum post](https://forum.seafile.com/t/synced-libraries-unavailable-after-changing-server-url/12916).

**Comment by killing** [comment]
@unDocUMeantIt Have you tried to upgrade to 7.0.9 version? You can use our official repository to install it. Please note that there was a change of repository servers recently.

**Comment by unDocUMeantIt** [comment]
i can install another version of the client, but i have worked around the problem manually, so there's no way for me to test another setup now that the new URL is finally configured. it took me several days to get everything up and running on various machines (i am not the only one using the server, everyone was having this problem), and i have no interest in getting into something like this ever again.

i took the time to document my findings, for you to examine what led to the issue. looking at the changelog of the client software, i didin't notice a fix that looked like it was related to this behaviour, so i would assume it is still not fixed and will bite the next one who tries to change their server's URL.

**Comment by killing** [comment]
We'll look into the issue.

**Comment by andrewramsay** [comment]
I've run into exactly this problem 2 or 3 times in the last few months when changing my server address. 

The first time it happened I may have been using an older client, but I've just had to do it again today with the 7.0.9 version (on Windows) so it is definitely still an issue. 

It should be very easy to reproduce:
- Left click avatar
- Select your account and then "Account settings"
- Enter new server address "https://SERVER:PORT" and click OK
- Restart the client and watch seafile.log for entries like this:

> [12/03/20 16:35:11] http-tx-mgr.c(783): libcurl failed to GET SERVER/seafhttp/protocol-version: Couldn't connect to server.

In other words, given a new server address like "https://seafile.example.com:12345" what actually ends up in repo.db is "seafile.example.com". 

I think this only affects repo.db, the server address entries in other databases like accounts.db and certs.db are updated correctly.

**Comment by cray12399** [comment]
I am trying to update my server address to https://(server)/seafile. How can I accomplish this?
